### PR TITLE
🧹 Validate safe config in hardhat.config.ts

### PR DESCRIPTION
**Pull request by ravinagill15**
## In this PR:
- https://layerzerolabs.atlassian.net/browse/APPS-85
- A task to validate all `safeConfigs` within the local `hardhat.config.ts` has been implemented within `devtools-evm-hardhat`
- A corresponding test file `validate.safe.configs.test.ts` has also been created.

### CODE REVIEW

File: examples/oft/hardhat.config.ts
Comment (ravinagill15) [inline examples/oft/hardhat.config.ts:93]: I can remove the safe configs here before merging -- just added here for quick testing. I obtained these safe addresses from the `ZROClaim` repo [here](https://github.com/LayerZero-Labs/ZROClaim/blob/d9acc7a2f54497670c71b11e47e098a688930d2c/hardhat.config.ts#L101-L122)

File: examples/oft/hardhat.config.ts
Comment (ravinagill15) [inline examples/oft/hardhat.config.ts:110]: See todo -- which api url is better for validating? I am assuming that it would be the general info api endpoint instead of this one that checks the balances. Also perhaps we should be using their `api/v2`?

**Comment by ravinagill15** [comment]
Please hold off on reviewing -- I will convert this into a task within `devtools-evm-hardhat` package

**Comment by ravinagill15** [comment]
Tests incoming!

File: packages/devtools-evm-hardhat/src/tasks/healthcheck/validate-safe-config.ts
Comment (ravinagill15) [inline packages/devtools-evm-hardhat/src/tasks/healthcheck/validate-safe-config.ts:20]: see todo

File: tests/devtools-evm-hardhat-test/test/task/validate.safe.configs.test.ts
Comment (ravinagill15) [inline tests/devtools-evm-hardhat-test/test/task/validate.safe.configs.test.ts:68]: issue with these tests -- the task always returns undefined, indicating perhaps I am calling the task incorrectly

File: packages/devtools-evm-hardhat/src/tasks/healthcheck/validate-safe-configs.ts
Comment (janjakubnanista) [inline packages/devtools-evm-hardhat/src/tasks/healthcheck/validate-safe-configs.ts:40]: `logger.error` is nicer

File: packages/devtools-evm-hardhat/src/tasks/healthcheck/validate-safe-configs.ts
Comment (janjakubnanista) [inline packages/devtools-evm-hardhat/src/tasks/healthcheck/validate-safe-configs.ts:43]: The same as above

File: tests/devtools-evm-hardhat-test/hardhat.config.with-invalid-safe-networks.ts
Comment (janjakubnanista) [inline tests/devtools-evm-hardhat-test/hardhat.config.with-invalid-safe-networks.ts:13]: Here you can do what you originally did for the `validHre` / `invalidHre` and just import the base config and add what you need. That way you don't need to have any of the imports and/or other settings duplicated

```typescript
// ...
import baseConfig from './hardhat.config';

const config: HardhatUserConfig = {
  ...baseConfig,
  networks: {
    ...baseConfig.networks,
    // ...
  }
}
}
```

File: tests/devtools-evm-hardhat-test/hardhat.config.with-missing-safe-address.ts
Comment (janjakubnanista) [inline tests/devtools-evm-hardhat-test/hardhat.config.with-missing-safe-address.ts:13]: Same here

File: tests/devtools-evm-hardhat-test/hardhat.config.with-missing-safe-url.ts
Comment (janjakubnanista) [inline tests/devtools-evm-hardhat-test/hardhat.config.with-missing-safe-url.ts:13]: Same here

File: tests/devtools-evm-hardhat-test/hardhat.config.with-valid-safe-networks.ts
Comment (janjakubnanista) [inline tests/devtools-evm-hardhat-test/hardhat.config.with-valid-safe-networks.ts:13]: Same here

**Review by janjakubnanista — CHANGES_REQUESTED** [review]
Nice!!! Just those couple tiny things and you're good. I think the tests will still not pass though because of the unrelated Solana test issue

**Review by janjakubnanista — APPROVED** [review]
Will need to be rebased on #873 and reapproved

**Pull request merged**
